packages/@aws-cdk/aws-route53/lib/cross-account-zone-delegation-handler/index.ts

```diff
@@ -15,7 +15,24 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
 
   switch (event.RequestType) {
     case 'Create':
+      return cfnEventHandler(resourceProps, false);
     case 'Update':
+      // We handle updates in two parts, first attempt to delete the old NS record, second create
+      // the new resource.
+
+      // DELETE old Resources
+      try {
+        const oldResourceProps = event.OldResourceProperties as unknown as ResourceProperties;
```

Comment:
I think we should only delete the old `NS` record when the `DelegatedZoneName` of the `OldResourceProperties` is different from the one from the (new) `ResourceProperties`. Otherwise we'll be removing and recreating the same record after every 'Update' event which is not necessary and might also produce outages.